### PR TITLE
Drop NFD gRPC API

### LINKED ISSUE #1546: Drop NFD gRPC API
**What would you like to be added**:

Remove the gRPC API that was deprecated in [v0.14.0](https://github.com/kubernetes-sigs/node-feature-discovery/releases/tag/v0.14.0) and replaced by NodeFeature CRD API. This includes code (including all related config and command line options like TLS), tests and documentation.

By our [deprecation policy](https://kubernetes-sigs.github.io/node-feature-discovery/v0.15/reference/versions.html) (I think this would fall under the "configuration options") we could do this already in v0.16 if so desired.

**Why is this needed**:

Drop deprecated feature. Simplify code, tests and documentation.

**Tasks**:

- [x] Flip the feature gate to GA (#1778)
- [x] Drop the `-enable-nodefeature-api` flag (#1780)
- [x] Drop the TLS-related flags (`-ca-file`, `-cert-file`, `-key-file`, `-verify-node-name`)
- [x] Helm: drop support for cert-manager
- [x] Drop the `-port` flag from nfd-master
- [x] Drop the `-server` and `-server-name-override` flags from nfd-worker
- [x] Make the NFD API controller perma-on, drop `-featurerules-controller` and `-crd-controller` flags
- [x] Drop the gRPC API (`pkg/labeler` package)
- [x] Update developer guide in the docs (running nfd-daemons locally, stand-alone, as described there is not possible anymore)
- [x] Drop all documentation related to the gRPC API

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Fixes #1546

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
Documentation drop is on PR https://github.com/kubernetes-sigs/node-feature-discovery/pull/1912

File: cmd/nfd-master/main.go [hunk 1]
```diff
@@ -122,17 +122,8 @@ func main() {
 func initFlags(flagset *flag.FlagSet) (*master.Args, *master.ConfigOverrideArgs) {
 	args := &master.Args{}
 
-	flagset.StringVar(&args.CaFile, "ca-file", "",
```
Comment (marquiz) [inline cmd/nfd-master/main.go:125]: Drop these from the flags switch-case above, too

File: pkg/nfd-worker/nfd-worker.go [hunk 2]
```diff
@@ -303,14 +295,6 @@ func (w *nfdWorker) Run() error {
 		return err
 	}
 
-	// Create watcher for TLS certificates
-	w.certWatch, err = utils.CreateFsWatcher(time.Second, w.args.CaFile, w.args.CertFile, w.args.KeyFile)
```
Comment (marquiz) [inline pkg/nfd-worker/nfd-worker.go:307]: Drop CaFile and friends from `args`, too (and the flags from main.go)

File: deployment/helm/node-feature-discovery/templates/master.yaml
Comment (marquiz) [inline deployment/helm/node-feature-discovery/templates/master.yaml:1]: There seems to be some stale tls leftovers in `templates/topologyupdater.yaml` as well. Drop those, too

File: deployment/helm/node-feature-discovery/values.yaml [hunk 3]
```diff
@@ -53,10 +53,6 @@ master:
     #   retryPeriod: 2s
     # nfdApiParallelism: 10
   ### <NFD-MASTER-CONF-END-DO-NOT-REMOVE>
-  # The TCP port that nfd-master listens for incoming requests. Default: 8080
-  # Deprecated this parameter is related to the deprecated gRPC API and will
-  # be removed with it in a future release
-  port: 8080
```
Comment (marquiz) [inline deployment/helm/node-feature-discovery/values.yaml:59]: Shouldn't we drop the cmdline flag in the same PR as well?

File: vendor/github.com/Microsoft/go-winio/.gitattributes
Comment (marquiz) [inline vendor/github.com/Microsoft/go-winio/.gitattributes:1]: Adding `vendor/`?? Let's not at least do it in this PR

**Review by marquiz — COMMENTED** [review]
Thanks @ArangoGutierrez for taking this effort. It's a good start. A few comments below.

Also a nit about the commit messages :roll_eyes: Let's specify [context](https://www.kubernetes.dev/docs/guide/pull-requests/#providing-additional-context) in format like `docs: xyzzy...` instead of bracketted `[Documentation]`

File: vendor/github.com/Microsoft/go-winio/.gitattributes
Comment (ArangoGutierrez) [reply vendor/github.com/Microsoft/go-winio/.gitattributes:1]: meh this was a fat finger, not intended. my bad

**Comment by ArangoGutierrez** [comment]
> Thanks @ArangoGutierrez for taking this effort. It's a good start. A few comments below.
> 
> Also a nit about the commit messages 🙄 Let's specify [context](https://www.kubernetes.dev/docs/guide/pull-requests/#providing-additional-context) in format like `docs: xyzzy...` instead of bracketted `[Documentation]`

dropped that commit, as I want documentation to be a stand-alone PR

File: pkg/nfd-worker/nfd-worker.go (hunk 2 above)
Comment (ArangoGutierrez) [reply pkg/nfd-worker/nfd-worker.go:307]: done

File: deployment/helm/node-feature-discovery/templates/master.yaml
Comment (ArangoGutierrez) [reply deployment/helm/node-feature-discovery/templates/master.yaml:1]: done

File: vendor/github.com/Microsoft/go-winio/.gitattributes
Comment (ArangoGutierrez) [reply vendor/github.com/Microsoft/go-winio/.gitattributes:1]: done

**Comment by ArangoGutierrez** [comment]
@marquiz PR is ready for review

File: cmd/nfd-master/main.go [hunk 4]
```diff
@@ -80,16 +80,6 @@ func main() {
 			args.Overrides.ResyncPeriod = overrides.ResyncPeriod
 		case "nfd-api-parallelism":
 			args.Overrides.NfdApiParallelism = overrides.NfdApiParallelism
-		case "ca-file":
-			klog.InfoS("-ca-file is deprecated, will be removed in a future release along with the deprecated gRPC API")
-		case "cert-file":
-			klog.InfoS("-cert-file is deprecated, will be removed in a future release along with the deprecated gRPC API")
-		case "key-file":
-			klog.InfoS("-key-file is deprecated, will be removed in a future release along with the deprecated gRPC API")
-		case "port":
```
Comment (marquiz) [inline cmd/nfd-master/main.go:89]: Drop the `-port` flag from below, too

File: pkg/nfd-worker/nfd-worker.go (hunk 2 above)
Comment (marquiz) [reply pkg/nfd-worker/nfd-worker.go:307]: Hmm, the stuff is still in `Args`. Alson the `Server` and `ServerNameOverride` should be dropped

File: pkg/nfd-worker/nfd-worker.go [hunk 5]
```diff
@@ -303,14 +277,6 @@ func (w *nfdWorker) Run() error {
...
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply pkg/nfd-worker/nfd-worker.go:307]: done

File: pkg/nfd-master/nfd-master.go [hunk 6]
```diff
@@ -122,15 +117,11 @@ type ConfigOverrideArgs struct {
...
-	KeyFile       string
-	Klog          map[string]*utils.KlogFlagVal
-	Kubeconfig    string
-	CrdController bool
-	Port          int
+	ConfigFile string
+	Instance   string
+	Klog       map[string]*utils.KlogFlagVal
+	Kubeconfig string
+	Port       int
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply pkg/nfd-master/nfd-master.go:124]: done

**Review by marquiz — APPROVED** [review]
Thank you @ArangoGutierrez for this effort. If/when we missed something let's drop those with follow-up PRs
/lgtm

**Pull request merged**